nucliadb_search/nucliadb_search/search/fetch.py

```diff
@@ -207,14 +221,170 @@ async def get_text_paragraph(result: ParagraphResult, kbid: str) -> str:
         logger.warn(
             f"{result.uuid} {field} {field_type_int} extracted_text does not exist on DB"
         )
-        return ""
+        return "", {}
+
+    splitted_text, positions = split_text(
+        extracted_text.text, query, highlight=highlight_split
+    )
+
+    return splitted_text, positions
 
+
+async def get_text_paragraph(
+    result: ParagraphResult,
+    kbid: str,
+    query: Optional[str] = None,
+    highlight_split: bool = False,
+    split: bool = False,
+) -> Tuple[str, Dict[str, List[Tuple[int, int]]]]:
```

Comment:
Suggestion: Can we use some Type Aliases here, to help future us reason about what is going on here? If i'm not wrong we can keep the returns as they are.